File: async-scala-client/src/main/scala/io/swagger/client/api/InvoiceContingencyApi.scala
package io.swagger.client.api

import io.swagger.client.model.Body_4
import io.swagger.client.model.Inline_response_200_3
import io.swagger.client.model.Message
import io.swagger.client.model.ValidationError
import com.wordnik.swagger.client._
import scala.concurrent.Future
import collection.mutable

class InvoiceContingencyApi(client: TransportClient, config: SwaggerConfig) extends ApiClient(client, config) {

  def invoiceDropContingency(state: String)(implicit reader: ClientResponseReader[Unit]): Future[Unit] = {
    // create path and map variables
    val path = (addFmt("/invoices/contingency/{state}")
        replaceAll ("\\{" + "state" + "\\}",state.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (state == null) throw new Exception("Missing required parameter 'state' when calling InvoiceContingencyApi->invoiceDropContingency")


    val resFuture = client.submit("DELETE", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def invoiceSetContingency(state: String,
      body: Body_4)(implicit reader: ClientResponseReader[Unit], writer: RequestWriter[Body_4]): Future[Unit] = {
    // create path and map variables
    val path = (addFmt("/invoices/contingency/{state}")
        replaceAll ("\\{" + "state" + "\\}",state.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (state == null) throw new Exception("Missing required parameter 'state' when calling InvoiceContingencyApi->invoiceSetContingency")

    if (body == null) throw new Exception("Missing required parameter 'body' when calling InvoiceContingencyApi->invoiceSetContingency")

    val resFuture = client.submit("PUT", path, queryParams.toMap, headerParams.toMap, writer.write(body))
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def invoiceVerifyContingency(state: String)(implicit reader: ClientResponseReader[Inline_response_200_3]): Future[Inline_response_200_3] = {
    // create path and map variables
    val path = (addFmt("/invoices/contingency/{state}")
        replaceAll ("\\{" + "state" + "\\}",state.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (state == null) throw new Exception("Missing required parameter 'state' when calling InvoiceContingencyApi->invoiceVerifyContingency")


    val resFuture = client.submit("GET", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }


}

File: CsharpDotNet2-client/src/main/CsharpDotNet2/IO/Swagger/Api/InvoiceContingencyApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IInvoiceContingencyApi
    {
        /// <summary>
        /// Invoice Drop Contingency per State Drop State in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param>
        /// <returns></returns>
        void InvoiceDropContingency (string state);
        /// <summary>
        /// Invoice Set Contingency per State Set State in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param>
        /// <param name="body">Set Contingency</param>
        /// <returns></returns>
        void InvoiceSetContingency (string state, Body4 body);
        /// <summary>
        /// Invoice Verify Contingency per State Check if State was in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param>
        /// <returns>InlineResponse2003</returns>
        InlineResponse2003 InvoiceVerifyContingency (string state);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class InvoiceContingencyApi : IInvoiceContingencyApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="InvoiceContingencyApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public InvoiceContingencyApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="InvoiceContingencyApi"/> class.
        /// </summary>
        /// <returns></returns>
        public InvoiceContingencyApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// Invoice Drop Contingency per State Drop State in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param> 
        /// <returns></returns>            
        public void InvoiceDropContingency (string state)
        {
            
            // verify the required parameter 'state' is set
            if (state == null) throw new ApiException(400, "Missing required parameter 'state' when calling InvoiceDropContingency");
            
    
            var path = "/invoices/contingency/{state}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "state" + "}", ApiClient.ParameterToString(state));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.DELETE, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceDropContingency: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceDropContingency: " + response.ErrorMessage, response.ErrorMessage);
    
            return;
        }
    
        /// <summary>
        /// Invoice Set Contingency per State Set State in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param> 
        /// <param name="body">Set Contingency</param> 
        /// <returns></returns>            
        public void InvoiceSetContingency (string state, Body4 body)
        {
            
            // verify the required parameter 'state' is set
            if (state == null) throw new ApiException(400, "Missing required parameter 'state' when calling InvoiceSetContingency");
            
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling InvoiceSetContingency");
            
    
            var path = "/invoices/contingency/{state}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "state" + "}", ApiClient.ParameterToString(state));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                postBody = ApiClient.Serialize(body); // http body (model) parameter
    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.PUT, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceSetContingency: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceSetContingency: " + response.ErrorMessage, response.ErrorMessage);
    
            return;
        }
    
        /// <summary>
        /// Invoice Verify Contingency per State Check if State was in Contingency
        /// </summary>
        /// <param name="state">Brazilian State</param> 
        /// <returns>InlineResponse2003</returns>            
        public InlineResponse2003 InvoiceVerifyContingency (string state)
        {
            
            // verify the required parameter 'state' is set
            if (state == null) throw new ApiException(400, "Missing required parameter 'state' when calling InvoiceVerifyContingency");
            
    
            var path = "/invoices/contingency/{state}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "state" + "}", ApiClient.ParameterToString(state));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceVerifyContingency: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling InvoiceVerifyContingency: " + response.ErrorMessage, response.ErrorMessage);
    
            return (InlineResponse2003) ApiClient.Deserialize(response.Content, typeof(InlineResponse2003), response.Headers);
        }
    
    }
}
